scapy/packet.py

```diff
@@ -127,6 +131,7 @@ def __init__(self, _pkt=b"", post_transform=None, _internal=0, _underlayer=None,
         self.fieldtype = {}
         self.packetfields = []
         self.payload = NoPayload()
+        self.post_inits = []
         self.init_fields()
```

Comment:
NOTE TO SELF: just order the fields following the fields_desc order during initialization. That will fix the problem way easier.

Use stg like
```
for f in fields_desc:
    v = kwargs.get(v, None)
    if v:
        self.setfieldval.....
```

Reply:
FTR, PR moved to https://github.com/secdev/scapy/pull/1550